### ISSUE TITLE
Waiting for promise inside get route

### ISSUE THREAD

**Issue by butera-simone** [jcubic/wayne#19]
I'm building an app that includes a service worker (that I'm trying to manage with Wayne) and a dedicated worker (that runs a lightning-fs database). The dedicated worker must communicate with the main thread and with the service worker via broadcast channels. Among many other things, the application opens new tabs that get content fetched from the database by the dedicated worker and served by the service worker.

Here's an example of how a non-Wayne service worker would fetch data from the database through the dedicated worker in my app:

```
const channel = new BroadcastChannel('my-channel');

self.addEventListener('fetch', event => {
    if (event.request.url.endsWith('/yo')) {
        event.respondWith(
            makeResponse()
        )
    }
});


const makeResponse = async function() {
    let x = await getMessage();
    let y = await new Response(x, {
        headers: {
            'Content-Type': 'text/html'
        }
    });
    return y
}

async function getMessage() {
    return new Promise((resolve) => {
        channel.addEventListener('message', (event) => {
            channel.removeEventListener('message', event.currentTarget);
            resolve(event.data);
        });
        channel.postMessage('/core/');
    });
}
```
I have code on the dedicated worker that responds to that message with the content of /core/ :

```
const channel = new BroadcastChannel('my-channel');

channel.onmessage = async function(event) {
      const ind = await fs.promises.readdir(event.data);
  channel.postMessage(ind);
}
```

This example works fine (at least as long the main page stays open, which is how the app is intended to be used). But when I try to fit this inside a Wayne app.get, I incur into problems. I tried formulating the code in different ways, but I've incurred into weird behavior (the most common one is the "/yo" page not loading at the first attempt, loading on reload, and then alternating between not-loading&loading at every reload; I've noticed that if I wait a few seconds after a failed reload, the next reload is a failure too: a failed reload guarantees a successful reload at the next attempt only when the next attempt happens immediately after the failure).

I've tried a few things, including experimenting with code that should make the page load forever (something that's pretty easy to accoplish with a normal bootstrapped service worker), but it didn't work inside app.get, the page loads immediately and I get either a reload of "./" or a successful load of "./yo" (as I said, often alternating between the two). This made me wonder if the library abstracts away some of the control one would usually have over the http response?

I've looked inside the Wayne code, and while I get a general idea of it (besides including path manipulation with regex, you register routes in the Wayne object that associate a path with a function, and there's a "fetch" event listener that returns a blob with metadata based on the res method called), I deemed it more efficient to ask here instead of trying to reconstruct the entire code line by line (that could still be my last option, if you tell me that for some reason Wayne does not include ways to wait for the message from the dedicated worker, but I still decided to use it as a library, I'd might need to edit it). I suspect there's probably a fairly simple solution to this?

**Comment by jcubic** [comment]
Can you maybe create a simple application that will reproduce the issue with Wayne, the exact steps to reproduce the issue, and the expected behavior? It's hard to follow written text since there are a lot of moving parts.

**Comment by jcubic** [comment]
Also, may I ask why you use a dedicated worker just to put lightning-fs there? Why don't you put the lightning-fs into Service Worker?

**Comment by butera-simone** [comment]
Sure, my app uses a lot of libraries but as soon as I have my PC in front of me I'll write a minimalist example that includes only Wayne and lightning-fs to show it to you. 

In the meantime, if it would be easy for you to write a simple snippet of how you'd create an app.get() route that sends a message via broadcast channel, waits for a response, and then includes that response into res.html(), that would probably solves everything because it's basically what I'm trying to do here, and I might just be forgetting something basic :p

Wrt the dedicated worker, it's a long story that includes other libraries for workers, if you're curious about it I can explain, but as far as this issue is concerned:

- yes I could use it inside Wayne and that's my backup plan
- I tried the same architecture (dedicated worker with lightning-fs + service worker with Wayne) without other libraries for workers and I caused the same bugs

**Comment by jcubic** [comment]
It was a bug, a really silly mistake. The code that handled routes added by users didn't have a return statement and the fallback was aways called when the code was async.

**Comment by jcubic** [comment]
The working demo was added to `gh-pages` branch.

**Comment by butera-simone** [comment]
It works now 👍👍
